lib/App/Netdisco/SSHCollector/Platform/IOS.pm

```diff
@@ -51,4 +67,40 @@ sub arpnip {
     return @arpentries;
 }
 
+sub macsuck {
+  my ($self, $hostlabel, $ssh, $args) = @_;
+  
+  debug "$hostlabel $$ macsuck()";
+  my $cmds = <<EOF;
+terminal length 0
+show interfaces description
+show mac address-table
+EOF
+  my @data = $ssh->capture({stdin_data => $cmds}); chomp @data;
+  if ($ssh->error) {
+    info "$hostlabel $$ error in SSH command " . $ssh->error;
+    return;
+  }
+
+  my $interfaces = {};
+  my $macentries = {};
+  my $re_interface_line = qr/^([a-z]+)([0-9\/\.]+)\s+(up|down|admin down)\s+(up|down)/i;
+  my $re_mac_line = qr/^\s*(All|[0-9]+)\s+([0-9a-f]{4}\.[0-9a-f]{4}\.[0-9a-f]{4})\s+\S+\s+([a-zA-Z]+)([0-9\/\.]+)/i;
+  my $port;
+  foreach (@data) {
+    if ($_ && /$re_interface_line/) {
+      $port = sprintf "%s%s", $if_name_map->{$1}, $2;
+      $interfaces->{$port} = {
+        up_admin => ($3 eq "admin down") ? "down" : "up",
+        up => ($3 eq "down") ? "down" : "up",
+      };
+    } elsif ($_ && /$re_mac_line/) {
+      $port = sprintf "%s%s", $if_name_map->{$3}, $4;
+      ++$macentries->{$1}->{$port}->{mac_as_ieee($2)};
```

Comment:
does this mean that VLAN can be "All" ? Netdisco probably needs zero instead.

Reply:
Currently, the output of `show mac address-table` indicate outputs for ALL...
```
ehc-mor-ert01#sh mac address-table
          Mac Address Table
-------------------------------------------

Vlan    Mac Address       Type        Ports
----    -----------       --------    -----
 All    0100.0ccc.cccc    STATIC      CPU
 All    0100.0ccc.cccd    STATIC      CPU
 All    0100.0ccc.ccce    STATIC      CPU
 All    0180.c200.0000    STATIC      CPU
 All    0180.c200.0001    STATIC      CPU
 All    0180.c200.0002    STATIC      CPU
 All    0180.c200.0003    STATIC      CPU
 All    0180.c200.0004    STATIC      CPU
 All    0180.c200.0005    STATIC      CPU
 All    0180.c200.0006    STATIC      CPU
 All    0180.c200.0007    STATIC      CPU
 All    0180.c200.0008    STATIC      CPU
 All    0180.c200.0009    STATIC      CPU
 All    0180.c200.000a    STATIC      CPU
 All    0180.c200.000b    STATIC      CPU
 All    0180.c200.000c    STATIC      CPU
 All    0180.c200.000d    STATIC      CPU
 All    0180.c200.000e    STATIC      CPU
 All    0180.c200.000f    STATIC      CPU
 All    0180.c200.0010    STATIC      CPU
  10    24f2.7fc7.5bbe    DYNAMIC     Gi0/1/0
  10    f04a.027c.5bf4    STATIC      CPU
```

I'm not sure that we need to keep these entries...
If they are not present in the SNMP output, I think a better idea should be to remove these MAC address of the records...

Reply:
Thanks for the example! very helpful. I'll leave "All" in but set it to VLAN 0, and later on in Netdisco there's sanity checking code on all MAC addresses so it'll be fine.